### PR TITLE
CI HW Testing Update

### PR DESCRIPTION
**Pull request by P-R-O-C-H-Y**
## Description of Change
Updated upload-artifact action to v3.
Edited build_dir path where the sketches are built in to + paths for upload-artifact actions
Added `if-no-files-found: error` to upload-artifact action

### CODE REVIEW

File: .github/scripts/sketch_utils.sh [hunk 1]
```diff
@@ -121,16 +121,16 @@ function build_sketch(){ # build_sketch <ide_path> <user_path> <path-to-ino> [ex
     if [ -n "$ARDUINO_BUILD_DIR" ]; then
         build_dir="$ARDUINO_BUILD_DIR"
     elif [ $len -eq 1 ]; then
-        # build_dir="$sketchdir/build"
-        build_dir="$HOME/.arduino/build.tmp"
+        build_dir="$sketchdir/build"
+        # build_dir="$HOME/.arduino/build.tmp"
```
Comment (me-no-dev) [inline .github/scripts/sketch_utils.sh:125]: different solution is needed here. It really is not nice to run CI locally and generate all those files in your  local fork. Maybe generate the artifacts elsewhere?

Comment (P-R-O-C-H-Y) [reply .github/scripts/sketch_utils.sh:125]: OK will try to make it different way :)

Comment (P-R-O-C-H-Y) [reply .github/scripts/sketch_utils.sh]: Is it ok now? Should be done the way you proposed to.

Comment (me-no-dev) [reply .github/scripts/sketch_utils.sh]: yup. that is better :)

**Review by me-no-dev — APPROVED** [review]

**Comment by mrengineer7777** [comment]
Does this also apply to branch 3.0.0?

**Comment by P-R-O-C-H-Y** [comment]
> Does this also apply to branch 3.0.0?

@mrengineer7777 yes, as this PR is fixing the CI HW testing. But for 3.0 there might be a some more changes for the CI.

**Pull request merged**